srml/grandpa/src/lib.rs

```diff
@@ -78,11 +78,11 @@ pub struct StoredPendingChange<N> {
 }
 
 impl<N: Decode> Decode for StoredPendingChange<N> {
-	fn decode<I: codec::Input>(value: &mut I) -> Option<Self> {
+	fn decode<I: codec::Input>(value: &mut I) -> core::result::Result<Self, Error> {
 		let old = OldStoredPendingChange::decode(value)?;
-		let forced = <Option<N>>::decode(value).unwrap_or(None);
+		let forced = <Option<N>>::decode(value)?;
```

Comment:
this was introduce to support old format.

But a failing decode of a value result in weird behavior on input: the some data could have been read and removed from input, thus the decode of `Vec<StoredPendingChange<N>>` with stored_pengind change in old format (without the option) will result in having the byte after first element removed and so a two element vec in old format will fail.

Reply:
I removed it this should be done in another PR if so